### PR TITLE
[11.0][MIG] project_department

### PR DESCRIPTION
**Pull request by ghost**
Depends on:
- [x] OCA/account-analytic#145

### CODE REVIEW

**Comment by ghost** [comment]
actually, I have no idea why `project_closing` gets tested on build
there must be something wrong with Travis itself since it tries to test `project_closing` without even installing it, so
```python
self._project1.toggle_active()
```
basically calls the `odoo.models.toggle_active()` on that record without running all of the logic in `project_closing`.

**Comment by simahawk** [comment]
@arkostyuk `project_closing` IS installed and tested as it's part of 11.0 branch. Travis' OCA test runner installs and runs tests for all the modules :wink: 

Here is initialized: https://travis-ci.org/OCA/project/jobs/322239926#L1087
To replicate the same behavior you should install all the modules that you see on that line.

Although, tests are already failing on 11.0 since [last commit ](https://travis-ci.org/OCA/project/jobs/322239926#L1087) ...which is a translation commit... so it's still strange.

File: project_department/LICENSE [hunk 1]
```diff
@@ -0,0 +1,18 @@
+OpenERP, Open Source Management Solution
```
Comment (simahawk) [inline project_department/LICENSE]: this file can be removed

File: project_department/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2014 Joël Grand-Guillaume (Camptocamp)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Project Department Categorization",
+    "version": "11.0.1.0.0",
+    "author": "Camptocamp, Daniel Reis, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "category": "Generic Modules/Projects & Services",
+    "website": "http://camptocamp.com",
```
Comment (simahawk) [inline project_department/__manifest__.py]: https://github.com/OCA/project

File: project_department/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "depends": [
+        "analytic_base_department",
+        "project",
+    ],
+    "data": ["views/project.xml"],
+    "auto_install": True,
```
Comment (simahawk) [inline project_department/__manifest__.py]: you can remove this flags

File: project_department/views/project.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    <record id="view_project_project_filter" model="ir.ui.view">
+         <field name="name">project.project.select</field>
+         <field name="model">project.project</field>
+         <field name="inherit_id" ref="project.view_project_project_filter"/>
+         <field name="type">search</field>
+         <field name="arch" type="xml">
+               <field name="partner_id" position="before">
+                  <field name="department_id"/>
+               </field>
+               <xpath expr="//filter[contains(@domain, user_id)]" position="after">
```
Comment (simahawk) [inline project_department/views/project.xml]: What are we selecting here? I don't see this filter in the orig view: https://github.com/odoo/odoo/blob/11.0/addons/project/views/project_views.xml#L175

File: project_department/views/project.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        </field>
+    </record>
+
+    <record id="view_task_tree2" model="ir.ui.view">
+        <field name="name">project.task.department.tree</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_tree2"/>
+        <field eval="2" name="priority"/>
+        <field name="arch" type="xml">
+            <field name="user_id" invisible="context.get('user_invisible', False)" position="after">
```
Comment (simahawk) [inline project_department/views/project.xml]: no need to replicate "invisible" attr

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@simahawk see https://github.com/OCA/project/issues/353

File: project_department/models/project.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    project_department_id = fields.Many2one(
+        related='project_id.department_id',
+        comodel_name='hr.department',
```
Comment (mpanarin) [inline project_department/models/project.py]: this line is redundant as it is already present in the field it relates to.

File: project_department/README.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Joël Grand-Guillaume <joel.grandguillaume@camptocamp.com>
+* Daniel Reis <dreis.pt@hotmail.com>
+* Yannick Vaucher <yannick.vaucher@camptocamp.com>
+* Dave Burkholder <dave@thinkwelldesigns.com>
```
Comment (mpanarin) [inline project_department/README.rst:43]: add contributor )

**Comment by simahawk** [comment]
@arkostyuk pls check https://github.com/OCA/project/issues/353 and go ahead w/ it

**Comment by simahawk** [comment]
rebase once this https://github.com/OCA/project/pull/356 is merged

**Comment by astirpe** [comment]
Travis tests relaunched, now it's all green.

**Comment by simahawk** [comment]
@astirpe tnx! @arkostyuk can you squash your commits?

**Comment by ghost** [comment]
@simahawk sure thing :)

**Review by asaunier — APPROVED** [review]
LGTM

* PR https://github.com/OCA/account-analytic/pull/145 the current PR depends on has been merged in the meantime.
* All requested changes have been done and commits have been squashed.

File: oca_dependencies.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+account-analytic https://github.com/arkostyuk/account-analytic 11.0-mig-analytic_base_department
```
Comment (simahawk) [inline oca_dependencies.txt]: @arkostyuk drop this

Comment (leemannd) [reply oca_dependencies.txt]: Don't we want to keep at least `account-analytic` or travis will fail?

**Review by leemannd — APPROVED** [review]
Approved once the question raised by @simahawk is answered.

**Review by simahawk — APPROVED** [review]

**Pull request merged**